File: src/Components/RecordItem.js
import { createElement } from "react";
import React from "react";
import { int32ToString, createObj } from "../fable_modules/fable-library-js.4.28.0/Util.js";
import { singleton, append, delay, toList } from "../fable_modules/fable-library-js.4.28.0/Seq.js";
import { reactApi } from "../fable_modules/Feliz.2.9.0/Interop.fs.js";
import { ofArray } from "../fable_modules/fable-library-js.4.28.0/List.js";
import { isNullOrWhiteSpace } from "../fable_modules/fable-library-js.4.28.0/String.js";
import { defaultOf } from "../fable_modules/fable-library-js.4.28.0/Util.js";

export function RecordItem(recordItemInputProps) {
    let elems_3;
    const onPhotoClick = recordItemInputProps.onPhotoClick;
    const onDelete = recordItemInputProps.onDelete;
    const onEdit = recordItemInputProps.onEdit;
    const currentUserId = recordItemInputProps.currentUserId;
    const record = recordItemInputProps.record;
    let timeDisplay;
    const matchValue = record.created_at;
    if (matchValue == null) {
        timeDisplay = "";
    }
    else {
        const timestamp = matchValue;
        timeDisplay = (new Date(timestamp).toLocaleTimeString('ko-KR', {hour: '2-digit', minute: '2-digit'}));
    }
    let recordTypeLabel;
    const matchValue_1 = record.record_type;
    recordTypeLabel = ((matchValue_1 === "workout") ? "운동" : ((matchValue_1 === "text") ? "메모" : ((matchValue_1 === "photo") ? "사진" : "기록")));
    const isOwner = record.user_id === currentUserId;
    return createElement("div", createObj(ofArray([["key", int32ToString(record.id)], ["className", "bg-white rounded-lg p-3 shadow-sm flex items-start gap-3"], (elems_3 = toList(delay(() => {
        let elems;
        return append(singleton(createElement("div", createObj(ofArray([["className", "flex-shrink-0 w-12 h-12 rounded-lg bg-indigo-100 flex items-center justify-center"], (elems = [createElement("span", {
            className: "text-xs font-semibold text-indigo-700",
            children: recordTypeLabel,
        })], ["children", reactApi.Children.toArray(Array.from(elems))])])))), delay(() => {
            let elems_1;
            return append(singleton(createElement("div", createObj(ofArray([["className", "flex-1 min-w-0"], (elems_1 = toList(delay(() => append(singleton(createElement("div", {
                className: "text-xs text-gray-500 mb-1",
                children: timeDisplay,
            })), delay(() => {
                let matchValue_2, text_1;
                return append((matchValue_2 = record.text_content, (matchValue_2 != null) ? (!isNullOrWhiteSpace(matchValue_2) ? ((text_1 = matchValue_2, singleton(createElement("div", {
                    className: "text-sm text-gray-800 whitespace-pre-wrap break-words",
                    children: text_1,
                })))) : singleton(defaultOf())) : singleton(defaultOf())), delay(() => {
                    const matchValue_3 = record.photo_url;
                    if (matchValue_3 == null) {
                        return singleton(defaultOf());
                    }
                    else {
                        const url = matchValue_3;
                        return singleton(createElement("img", {
                            src: url,
                            alt: "운동 사진",
                            className: "w-16 h-16 object-cover rounded mt-1 cursor-pointer hover:opacity-80 transition-opacity",
                            onClick: (_arg) => {
                                onPhotoClick(url);
                            },
                        }));
                    }
                }));
            })))), ["children", reactApi.Children.toArray(Array.from(elems_1))])])))), delay(() => {
                let elems_2;
                return isOwner ? singleton(createElement("div", createObj(ofArray([["className", "flex-shrink-0 flex items-start gap-2"], (elems_2 = [createElement("button", {
                    onClick: (_arg_1) => {
                        onEdit(record.id);
                    },
                    className: "text-sm text-blue-600 hover:text-blue-800 transition-colors",
                    children: "수정",
                }), createElement("button", {
                    onClick: (_arg_2) => {
                        onDelete(record.id);
                    },
                    className: "text-sm text-red-600 hover:text-red-800 transition-colors",
                    children: "삭제",
                })], ["children", reactApi.Children.toArray(Array.from(elems_2))])])))) : singleton(defaultOf());
            }));
        }));
    })), ["children", reactApi.Children.toArray(Array.from(elems_3))])])));
}

File: src/Components/RecordItem.fs
module Components.RecordItem

open Feliz
open Fable.Core
open Fable.Core.JsInterop
open Supabase.Types

[<ReactComponent>]
let RecordItem (record: WorkoutRecord) (currentUserId: string) (onEdit: int -> unit) (onDelete: int -> unit) (onPhotoClick: string -> unit) =
    // Format time from created_at
    let timeDisplay =
        match record.created_at with
        | Some timestamp ->
            emitJsExpr timestamp "new Date($0).toLocaleTimeString('ko-KR', {hour: '2-digit', minute: '2-digit'})"
        | None -> ""

    // Get record type label
    let recordTypeLabel =
        match record.record_type with
        | "workout" -> "운동"
        | "text" -> "메모"
        | "photo" -> "사진"
        | _ -> "기록"

    // Check if current user owns this record
    let isOwner = record.user_id = currentUserId

    Html.div [
        prop.key (string record.id)
        prop.className "bg-white rounded-lg p-3 shadow-sm flex items-start gap-3"
        prop.children [
            // Left side: Type icon/label
            Html.div [
                prop.className "flex-shrink-0 w-12 h-12 rounded-lg bg-indigo-100 flex items-center justify-center"
                prop.children [
                    Html.span [
                        prop.className "text-xs font-semibold text-indigo-700"
                        prop.text recordTypeLabel
                    ]
                ]
            ]

            // Center: Content area
            Html.div [
                prop.className "flex-1 min-w-0"
                prop.children [
                    // Time display
                    Html.div [
                        prop.className "text-xs text-gray-500 mb-1"
                        prop.text timeDisplay
                    ]

                    // Text content (if present)
                    match record.text_content with
                    | Some text when not (System.String.IsNullOrWhiteSpace(text)) ->
                        Html.div [
                            prop.className "text-sm text-gray-800 whitespace-pre-wrap break-words"
                            prop.text text
                        ]
                    | _ -> Html.none

                    // Photo thumbnail (if present)
                    match record.photo_url with
                    | Some url ->
                        Html.img [
                            prop.src url
                            prop.alt "운동 사진"
                            prop.className "w-16 h-16 object-cover rounded mt-1 cursor-pointer hover:opacity-80 transition-opacity"
                            prop.onClick (fun _ -> onPhotoClick url)
                        ]
                    | None -> Html.none
                ]
            ]

            // Right side: Edit/Delete buttons (owner only)
            if isOwner then
                Html.div [
                    prop.className "flex-shrink-0 flex items-start gap-2"
                    prop.children [
                        // Edit button
                        Html.button [
                            prop.onClick (fun _ -> onEdit record.id)
                            prop.className "text-sm text-blue-600 hover:text-blue-800 transition-colors"
                            prop.text "수정"
                        ]

                        // Delete button
                        Html.button [
                            prop.onClick (fun _ -> onDelete record.id)
                            prop.className "text-sm text-red-600 hover:text-red-800 transition-colors"
                            prop.text "삭제"
                        ]
                    ]
                ]
            else
                Html.none
        ]
    ]
